onnxruntime/core/providers/mkldnn/math/sum.cc

```diff
@@ -0,0 +1,224 @@
+// Copyright(C) 2018 Intel Corporation
+// Licensed under the MIT License
+
+#ifdef _WIN32
+#pragma warning(disable : 4244)
+#endif
+
+#include "core/providers/mkldnn/mkldnn_common.h"
+#include "core/providers/mkldnn/math/sum.h"
+#include "core/providers/mkldnn/mkldnn_fwd.h"
+
+namespace onnxruntime {
+namespace mkl_dnn {
+
+namespace {
+// Struct which encapsulates parameters for MKLDNN Sum primitives.
+struct SumParams {
+  std::vector<mkldnn::memory::dims>& src_dims;
```

Comment:
is there any reason these should be just ref and not const ref? same comment for dst_dim.